### PR TITLE
Customer orders eric

### PR DESCRIPTION
**Pull request by eknigge**
### Front-End
1. Vendor order page now includes react routing. Uses id in url to show cart orders.
2. Updated customer menu page to use react routing. 
3. Customer menu page now includes ability to submit order. I was able to get the format of the data correct, but I'm running into a bug where the console log shows the correct output, but the PUT request is a blank body. 
4. Fixed customer ordering page to include name of item being ordered.

### Back-End
1. Updated customer menu put request. Accepts JSON data in the format provided in the testdata.json file. Will update the API once the front-end and back-end are linked.
2. Added testdata.json and python test script for put method. 
3. Updated database table ITEM_MENU AVAILABLE to AVAILABLE_MENU so that this field is not ambiguous. 
4. Updated customer menu query so that only items that are marked as availableare shown.

### CODE REVIEW

File: Docs/Database/CREATE_DATABASE.sql [hunk 1]
```diff
@@ -47,7 +47,7 @@ ENGINE = InnoDB;
 CREATE TABLE IF NOT EXISTS `mydb`.`ITEMS_MENU` (
   `ITEM_ID` INT NOT NULL,
   `MENU_ID` INT NOT NULL,
-  `AVAILABLE` ENUM("Y", "N") NOT NULL,
+  `AVAILABLE_MENU` ENUM("Y", "N") NOT NULL,
```
Comment (shipitsteven) [inline Docs/Database/CREATE_DATABASE.sql:50]: Making changes to the database this late in the development means we have to hunt down every query call to this column and update its name. 
If it's conflicting with `Cart.Available`, you can specify which column you're calling in the SQL statement like `Items_Menu.Available`. Better to fix one place than N places.

File: front-end/hot-dog/src/CustomerComponents/CustomerMenu.js [hunk 2]
```diff
@@ -7,12 +7,34 @@ class CustomerMenu extends React.Component{
...
+        for(let key in this.state){
+            order[key] = this.state[key]
+        }
+        let orderJSON = {
+            "order": order,
+            "cartID":this.props.cartID,
+            "userID":2
+        }
+
+        fetch(`http://localhost:5000/customer/order/`, {
```
Comment (shipitsteven) [inline front-end/hot-dog/src/CustomerComponents/CustomerMenu.js:22]: To send a JSON as the body of the request, you would need to specify a header like `headers: { 'Content-Type': 'application/json' }` **and** stringify the JSON object. These two reasons are probably what's causing the `undefined` behavior in the back-end.

Example [line 50-53]: https://github.com/eknigge/AD320_Project/blob/main/front-end/hot-dog/src/components/TableRow.js

Source: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_json_data

**Review by shipitsteven — CHANGES_REQUESTED** [review]
Good work Eric, I left a couple notes in the files that may help you solve your problem. There's also a suggestion on the DB modification, I know that the vendor menu page rely on that column for a few queries.

File: Docs/Database/CREATE_DATABASE.sql (hunk 1 above)
Comment (eknigge) [reply Docs/Database/CREATE_DATABASE.sql:50]: Agree, I'm rusty on SQL. That's an easier and simpler solution. I'll revert the change and update the request.

**Comment by eknigge** [comment]
Thanks for the review and help! I was able to resolve the merge conflicts and got the ordering system to work. There's no indication and than order has been submitted, but it works when you run a test. Not sure if we want to create an alert or modal, but the base functionality is present, which is the most important thing.

File: front-end/hot-dog/src/CustomerComponents/CustomerMenu.js [hunk 3]
```diff
@@ -7,12 +7,36 @@ class CustomerMenu extends React.Component{
...
+            method:'PUT',
+            body:orderJSON,
```
Comment (shipitsteven) [inline front-end/hot-dog/src/CustomerComponents/CustomerMenu.js:24]: Is this a duplicate `body`? It's okay if the code is working for now, might want to delete it in future versions

**Review by shipitsteven — APPROVED** [review]
Good work Eric! Glad you're able to finish the POST request. Another feature done front-to-back 🎉💯

**Pull request merged**